shell.nix

```diff
@@ -0,0 +1,25 @@
+let
+  nixpkgs = import <nixpkgs> {};
+  make-monero = nixpkgs.writeShellScriptBin "make-monero" "make -j$NIX_BUILD_CORES";
+in
+with nixpkgs;
+stdenv.mkDerivation rec {
+  name = "monero-build-environment";
+
+  buildInputs = [
+    cmake
+    git
+    gcc
+    pkgconfig
+    boost
+    openssl
+    zeromq3
+    cppzmq
+    unbound
+    libsodium
+    # needed for ledger nano s
+    hidapi
```

Comment:
also needs protobuf and libusb , if you want Trezor support as well (though optional, like hidapi).